api.go

```diff
@@ -223,47 +386,53 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 		logger = log.New(conf.LogOutput, "", log.LstdFlags)
 	}
 
-	// Try to restore the current term
+	// Try to restore the current term.
 	currentTerm, err := stable.GetUint64(keyCurrentTerm)
 	if err != nil && err.Error() != "not found" {
 		return nil, fmt.Errorf("failed to load current term: %v", err)
 	}
 
-	// Read the last log value
-	lastIdx, err := logs.LastIndex()
+	// Read the last log value.
+	lastIndex, err := logs.LastIndex()
 	if err != nil {
 		return nil, fmt.Errorf("failed to find last log: %v", err)
 	}
 
-	// Get the log
+	// Get the log.
 	var lastLog Log
-	if lastIdx > 0 {
-		if err = logs.GetLog(lastIdx, &lastLog); err != nil {
-			return nil, fmt.Errorf("failed to get last log: %v", err)
+	if lastIndex > 0 {
+		if err = logs.GetLog(lastIndex, &lastLog); err != nil {
+			return nil, fmt.Errorf("failed to get last log at index %d: %v", lastIndex, err)
 		}
 	}
 
+	// Make sure we have a valid server address and ID.
+	protocolVersion := conf.ProtocolVersion
 	localAddr := ServerAddress(trans.LocalAddr())
 	localID := conf.LocalID
-	if localID == "" {
-		logger.Printf("[WARN] raft: No server ID given, using network address: %v. This default will be removed in the future. Set server ID explicitly in config.",
+
+	// TODO (slackpad) - When we deprecate protocol version 1, remove this
+	// along with the AddPeer() and RemovePeer() APIs.
+	if protocolVersion < 2 {
+		logger.Printf("[INFO] raft: Configured ProtocolVersion < 2, using network address as LocalID: %v",
```

Comment:
The way it's set up now, they can start setting the `LocalID` right away, and then they get an INFO-level message when things start up saying "we are using the network address as the LocalID". I figured this was safest so we don't require them to do something special in the config to make things match.